src/domain.js

```diff
@@ -0,0 +1,73 @@
+/**
+ * Domains define the rules in which resolved actions are converted into
+ * new state. They are added to a Microcosm instance using `addDomain`.
+ * @constructor
+ * @param {Microcosm} repo The instance of Microcosm that added this domain.
+ * @param {Object} [options] Options passed from `addDomain`
+ */
+export default function Domain (options, repo) {
+  this.options = options
+  this.repo = repo
```

Comment:
my js-fu isn't strong enough to understand how defining this function does anything, but does the fact that the order is `options, repo` here and `repo, options` in `setup` have any significance?